tw-tasks-core/src/main/java/com/transferwise/tasks/entrypoints/MdcService.java

```diff
@@ -0,0 +1,112 @@
+package com.transferwise.tasks.entrypoints;
+
+import com.newrelic.api.agent.NewRelic;
+import com.transferwise.common.baseutils.ExceptionUtils;
+import com.transferwise.common.context.TwContext;
+import com.transferwise.tasks.TasksProperties;
+import com.transferwise.tasks.domain.IBaseTask;
+import com.transferwise.tasks.domain.ITask;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.Callable;
+import lombok.NonNull;
+import org.slf4j.MDC;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class MdcService implements IMdcService {
+
+  private static final ThreadLocal<Set<String>> keysSet = new ThreadLocal<>();
+
+  @Autowired
+  private TasksProperties tasksProperties;
```

Comment:
Since only MDC is used from the properties maybe worth using it instead so that we don't need to call getter every time

Reply:
I thought about it as well. But I think JIT is taking care of that and will inline it at one point anyway.